pkg/auth/ldaputil/url.go

```diff
@@ -246,3 +246,12 @@ func DetermineDerefAliasesBehavior(derefAliasesString string) (DerefAliases, err
 	}
 	return derefAliases, nil
 }
+
+func ExtractLDAPHost(url string) (string, error) {
```

Comment:
I don't like this.  I think I'd rather see an `Host() string` method on the `ldapclient.Config` type.